python/rmm/__init__.py

```diff
@@ -18,6 +18,7 @@
 from rmm.rmm import (
     RMMError,
     _make_finalizer,
+    _register_finalize,
```

Comment:
Lets rename to `_register_atexit_finalize` to not confuse it with the numba finalizer above.